src/js/helpers/useToast.ts

```diff
@@ -0,0 +1,172 @@
+  const toastTemplate = toastContainer.querySelector<HTMLTemplateElement>(selectorsMap.toast.template);
+  const toastClone = toastTemplate?.content.cloneNode(true) as DocumentFragment;
+  const toastElement = toastClone?.querySelector<HTMLElement>(selectorsMap.toast.toast);
+  const toastBody = toastElement?.querySelector<HTMLElement>(selectorsMap.toast.body);
+
+  if (toastElement && toastBody) {
+    toastContainer.appendChild(toastElement);
+
+    return toastElement;
+  }
+}
+
+const useFallbackToastTemplate = (toastContainer: HTMLElement): HTMLElement => {
+  toastContainer.innerHTML = '';
+  const fallbackContainer = getFallbackContainer();
+
+  if (fallbackContainer) {
+    toastContainer.innerHTML = fallbackContainer.innerHTML;
+    const toastElement = cloneToastTemplate(toastContainer);
+
+    if (toastElement) {
+
+      return toastElement;
+    } 
+  }
+
+  throw new DOMException('The object can not be cloned.');
+}
+
+const useFallbackToastContainer = (): HTMLElement => {
+  const body = document.querySelector<HTMLBodyElement>('body');
+  const fallbackContainer = getFallbackContainer();
+
+  if (body && fallbackContainer) {
+    const toastElement = cloneToastTemplate(fallbackContainer);
+
+    if (toastElement) {
+      fallbackContainer.appendChild(toastElement);
+      body.appendChild(fallbackContainer);
+  
+      return toastElement;
+    }
+  }
+
+  throw new DOMException('The object can not be cloned.');
+}
+
+const getFallbackContainer = () => {
+  const dummyElement = document.createElement('div');
+  dummyElement.innerHTML = Toaster.Fallback;
+  const fallbackContainer =  dummyElement.querySelector<HTMLElement>(selectorsMap.toast.container);
+
+  return fallbackContainer;
+}
+
+const insertToastMessage = (toastElement: HTMLElement, message: string) => {
+  const toastBody = toastElement.querySelector<HTMLElement>(selectorsMap.toast.body);
+
+  if (toastBody) {
+    toastBody.innerHTML = message;
+  }
+}
+
+const addToastClassList = (toastElement: HTMLElement, options: Toaster.Option) => {
+  let bsClassList: string = Toaster.Theme[options.type];
+  const customClassList = options.classlist;
+  
+  if (customClassList) {
+    bsClassList = bsClassList.concat(' ', customClassList.trim());
+  }
+
+  bsClassList.split(' ').forEach((value) => {
+    if (value) {
+      toastElement.classList.add(value);
+    }
+  });
+}
+
+const setCloseButtonVisible = (toastElement: HTMLElement) => {
+  const closeButton = toastElement.querySelector<HTMLElement>(selectorsMap.toast.close);
+
+  if (closeButton) {
+    let toastColor = toastElement.classList.toString().match(/text-\w+/)?.toString();
+
+    if (toastColor) {
+      toastColor = toastColor.substring(toastColor.indexOf('-') + 1);
+
+      if (toastColor === 'white' || toastColor === 'light') {
+        closeButton.classList.add('btn-close-white');
+      }
+    }
+
+    closeButton.classList.remove('d-none');
+  }
+}
+
+export default useToast;
```

Comment:
As it is mature enought, let's not add `useToast` to `window.prestashop.components` so every modules and child themes will be able to use it without having a JS transpiler